target/linux/ath79/dts/qca9558_librerouter-v1.dts

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca9557.dtsi"
+
+/ {
+	compatible = "librerouter,librerouter-v1", "qca,qca9557";
+	model = "LibreRouter v1";
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	aliases {
+		led-boot = &system;
+		led-failsafe = &system;
+		led-running = &system;
+		led-upgrade = &system;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		system: system {
+			label = "librerouter:green:system";
+			gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "heartbeat";
+		};
+
+		wifi_green {
+			label = "librerouter:green:wlan2g";
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		gpio_poe_passthrough {
+			gpio-export,name = "librerouter:power:poe_pass";
+			gpio-export,output = <0>;
+			gpios = <&gpio 1 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	watchdog {
+		compatible = "linux,wdt-gpio";
+		gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+		hw_algo = "toggle";
+		hw_margin_ms = <1000>;
+		always-running;
+	};
+
+};
+
+&pcie0 {
+	status = "okay";
```

Comment:
Could you please add the pci child nodes for the attached AR9582?

(I think the qca9558_tplink_tl-wdr4900-v2.dts could be a good example. It has a AR9580 but the IDs should be similar)

(Also make sure that the reg values are good if you are unsure please provide a bootlog)

Reply:
The owner of the board can change one or both of the m-pcie boards, let's say to use a 3g-modem, to change the functionality of the device. 
If we add the pci child nodes a new dts file will be neded ? What do you think it's the best way to handle this?

Reply:
Do you know if the AR9582 are preinstalled when the device "leaves the factory"? If yes, then I would include the nodes. 

Note: The pci DT enumeration works differently and the pcie subsystem code is able to handle cases where the pci device was replaced by something else.

Reply:
Yes, they are preinstalled from the factory.  Ok I didn't know that the pcie subsystem can handle replacements, I will test it ;)

Reply:
@chunkeey As far as I understand calibration and mac for this mpci cards should be read from EEPROM. Searching how to do that I found this commit https://github.com/torvalds/linux/commit/ce938231bd3b1d7af3cbd8836f084801090470e1 so If I understand correctly not using `qca,no-eeprom` is broken in 4.14?

dmesg extract:
```
[    0.000000] Linux version 4.14.95 (san@repo) (gcc version 7.4.0 (OpenWrt GCC 7.4.0 r0+9138-09b5402)) #0 Sun Jan 27 14:56:16 2019
...
[    0.189522] PCI host bridge to bus 0000:00
[    0.193887] pci_bus 0000:00: root bus resource [mem 0x10000000-0x13ffffff]
[    0.201234] pci_bus 0000:00: root bus resource [io  0x0000]
[    0.207155] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
[    0.214378] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    0.222828] pci 0000:00:00.0: [168c:0033] type 00 class 0x028000
[    0.222881] pci 0000:00:00.0: reg 0x10: [mem 0x00000000-0x0001ffff 64bit]
[    0.222935] pci 0000:00:00.0: reg 0x30: [mem 0x00000000-0x0000ffff pref]
[    0.223009] pci 0000:00:00.0: supports D1
[    0.223017] pci 0000:00:00.0: PME# supported from D0 D1 D3hot
[    0.223220] pci 0000:00:00.0: BAR 0: assigned [mem 0x10000000-0x1001ffff 64bit]
[    0.231039] pci 0000:00:00.0: BAR 6: assigned [mem 0x10020000-0x1002ffff pref] 
...
[   11.002141] PCI: Enabling device 0000:00:00.0 (0000 -> 0002)             
[   11.099047] ath: EEPROM regdomain: 0x0                                   
[   11.099054] ath: EEPROM indicates default country code should be used    
[   11.099057] ath: doing EEPROM country->regdmn map search                 
[   11.099070] ath: country maps to regdmn code: 0x3a                       
[   11.099075] ath: Country alpha2 being used: US                           
[   11.099078] ath: Regpair used: 0x3a                                      
[   11.109284] ieee80211 phy1: Selected rate control algorithm 'minstrel_ht'
[   11.110968] ieee80211 phy1: Atheros AR9300 Rev:4 mem=0xb0000000, irq=18  
```

Reply:
> I understand correctly not using qca,no-eeprom is broken in 4.14?

That issue is fixed in Openwrt's mac80211 package. As for the vanilla 4.14, yes this would be a problem there... But I did tell stable@vger.kernel.org to add that patch to a upcoming 4.14 stable release.

Reply:
Great, it works! :)